### DISCUSSION TITLE
What specific order does RegisterCommandsAsync follow in order to load commands?

### DISCUSSION THREAD

**Discussion by ghost** [VsixCommunity/Community.VisualStudio.Toolkit#525] [Q&A] [answered]
I need one command to be registered the last, since it dinamically changes some other commands state. If this command is loaded first and tries to access other commands, it returns a null exception. However, I can't figure the order in which the commands are being loaded. I tried changing the priority in the .vsct file, but it is not working. Any suggestions?

**Comment by reduckted** [comment, marked as answer]
`RegisterCommandsAsync` is just a shortcut for calling `InitializeAsync` on each command type. If you need the commands to be registered in a specific order, you can register them manually in the desired order.

```diff
-await this.RegisterCommandsAsync();
+await Command1.InitializeAsync(this);
+await Command2.InitializeAsync(this);
+await Command3.InitializeAsync(this);
+await Command4.InitializeAsync(this);
```

**Reply by ghost** [reply]
Oh, didn't think of that. I guess it will do. RegisterCommandsAsync is too unpredictable with my current approach. Thanks!
